libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -108,71 +111,80 @@ func New(cfg *common.Config) (processors.Processor, error) {
 		kubernetesAvailable: false,
 	}
 
-	client, err := kubernetes.GetKubernetesClient(config.KubeConfig)
-	if err != nil {
-		if kubernetes.IsInCluster(config.KubeConfig) {
-			log.Debugf("Could not create kubernetes client using in_cluster config: %+v", err)
-		} else if config.KubeConfig == "" {
-			log.Debugf("Could not create kubernetes client using config: %v: %+v", os.Getenv("KUBECONFIG"), err)
-		} else {
-			log.Debugf("Could not create kubernetes client using config: %v: %+v", config.KubeConfig, err)
+	// complete processor's initialisation asynchronously so as to re-try on failing k8s client initialisations in case
+	// the k8s node is not yet ready.
+	go func(processor *kubernetesAnnotator) {
```

Comment:
In other processors this code lives in a [`processor.init()` function](https://github.com/elastic/beats/blob/c5a568f761b63414f3415e8f6eceb13b84919dbf/libbeat/processors/add_cloud_metadata/add_cloud_metadata.go#L76), we might do the same here.
```suggestion
	go processor.init(config)
```

Reply:
Thanks for pointing to this!